src/software/jetson_nano/thunderloop.cpp

```diff
@@ -434,4 +458,11 @@ void Thunderloop::updateErrorCodes()
     {
         robot_status_.mutable_error_code()->Add(TbotsProto::ErrorCode::HIGH_BOARD_TEMP);
     }
+
+    std::ifstream log_file(PATH_TO_RINGBUFFER_LOG);
+    if (!isPowerStable(log_file))
+    {
+        robot_status_.mutable_error_code()->Add(
+            TbotsProto::ErrorCode::UNSTABLE_POWER_SUPPLY);
+    }
```

Comment:
Per our conversation a few weeks back, I think it would be better to open the file in the constructor of `Thunderloop` so we don't have to search from the beginning of the logs everytime.

Reply:
I feel like a static variable would make more sense. Since this is only accessed in one place, which is this one function.